### ISSUE TITLE
Failed to load dynamic library 'JavaScriptCore.framework/JavaScriptCore'

### ISSUE THREAD

**Issue by narumi147** [abner/flutter_js#126]
**Versions**:
platform: macOS release (crash report collected from user), some devices.

**Error**:
```
Failed to load dynamic library 'JavaScriptCore.framework/JavaScriptCore': dlopen(JavaScriptCore.framework/JavaScriptCore, 1): no suitable image found.  Did find:
	file system relative paths not allowed in hardened programs
```

**StackTrace**:

Caused by
```dart
  DynamicLibrary.open('JavaScriptCore.framework/JavaScriptCore')
```

```
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22)
#2      JscFfi.lib (package:flutter_js/javascriptcore/binding/jsc_ffi.dart:7)
#3      jSContextGroupCreate (package:flutter_js/javascriptcore/binding/js_context_ref.dart)
#4      new JavascriptCoreRuntime (package:flutter_js/javascriptcore/jscore_runtime.dart)
#5      getJavascriptRuntime (package:flutter_js/flutter_js.dart:45)
```

**Others**:
According to [this blog](https://www.mbsplugins.de/archive/2021-07-13/Common_problems_loading_a_dyna/monkeybreadsoftware_blog_filemaker_)

> The file path misses the / on the front and thus is a relative path. But that is forbidden for library loading for applications using hardened runtime, which is more strict for improved security.

**Comment by hpelitebook745G2** [comment]
i'm also getting this, any updates?

**Comment by guuguo** [comment]
the same problem

**Comment by DeanFs** [comment]
I also encountered this problem. It seems that this library is no longer maintained? It's been over a year and still hasn't been closed.

**Comment by TaurusXi** [comment]
Is there a solution ？

**Comment by abner** [comment]
https://github.com/flutter/flutter/issues/149407

I will change it and I will submit a patch version with the change

**Comment by abner** [comment]
Fixed in `flutter_js: ^0.8.1`
